### PR TITLE
Alerting: Support deleting rule groups in the provisioning API

### LINKED ISSUE #81860: Provisioning API: Support deleting a rule group
**Why is this needed**:

The Provisioning API can be used to create or update rule groups using a PUT request. However, rule groups cannot be DELETEd in a single call. To remove a rule group, it seems one must either DELETE each individual rule one-by-one, at which point the group is also deleted.

**What would you like to be added**:

Support calling DELETE on the `/api/v1/provisioning/{FOLDERUID}/rule-groups/{GROUPNAME}` endpoint. This should delete both the group and all of its component rules.

This is considerably more intuitive than having the rule group disappear after the rules are deleted one-by-one.

**Who is this feature for?**

All users of the provisioning API who deal with alert groups.

### PR DESCRIPTION
**Pull request by joeblubaugh**
Adds support for DELETE to the provisioning API's alert rule groups route, which allows deleting the rule group with a single API call. Previously, groups were deleted by deleting rules one-by-one.

Fixes #81860

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by joeblubaugh** [comment]
Could someone please help me generate the updated swagger documentation for this branch? go-swagger is panicking on me and I can't seem to track the problem down.

```❯ make
(re)installing /Users/joe/go/bin/swagger-v0.30.2
go: downloading github.com/go-swagger/go-swagger v0.30.2
go: downloading github.com/go-openapi/runtime v0.24.1
go: downloading github.com/toqueteos/webbrowser v1.2.0
go: downloading golang.org/x/sys v0.0.0-20220829200755-d48e67d00261
go: downloading github.com/go-openapi/inflect v0.19.0
go: downloading golang.org/x/tools v0.1.12
go: downloading go.mongodb.org/mongo-driver v1.10.1
go: downloading github.com/go-openapi/jsonreference v0.20.0
go: downloading github.com/google/uuid v1.1.2
go: downloading golang.org/x/crypto v0.0.0-20220829220503-c86fa9a7ed90
go: downloading github.com/subosito/gotenv v1.3.0
go: downloading gopkg.in/ini.v1 v1.66.4
go: downloading github.com/pelletier/go-toml/v2 v2.0.1
go: downloading golang.org/x/mod v0.6.0-dev.0.20220419223038-86c51ed26bb4
SWAGGER_GENERATE_EXTENSION=false /Users/joe/go/bin/swagger-v0.30.2 generate spec -m -w definitions -o spec.json && SWAGGER_GENERATE_EXTENSION=false /Users/joe/go/bin/swagger-v0.30.2 generate spec -m --include-tag=stable -o spec-stable.json
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1029b1dd0]

goroutine 30 [running]:
go/types.(*Checker).handleBailout(0x140004c6200, 0x14000dd3c38)
        /opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/check.go:367 +0x9c
panic({0x102e7dce0?, 0x1034e0ef0?})
        /opt/homebrew/Cellar/go/1.22.0/libexec/src/runtime/panic.go:770 +0x124
go/types.(*StdSizes).Sizeof(0x0, {0x102f6d3f8, 0x1034ebd40})
        /opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/sizes.go:228 +0x320
go/types.(*Config).sizeof(...)
```

**Review by yuri-tceretian — APPROVED** [review]
LGTM

File: pkg/services/ngalert/api/authorization_test.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func TestAuthorize(t *testing.T) {
 		}
 		paths[p] = methods
 	}
-	require.Len(t, paths, 64)
+	require.Len(t, paths, 65)
```
Comment (rwwiv) [inline pkg/services/ngalert/api/authorization_test.go:43]: This should have been bumped, but the test is failing.

**Review by rwwiv — APPROVED** [review]
LGTM, there's a test failure due to path count but looks like the all the added tests are succeeding.

File: pkg/services/ngalert/api/authorization_test.go (hunk 1 above)
Comment (joeblubaugh) [reply pkg/services/ngalert/api/authorization_test.go:43]: The output looks like this: 
```/ruler/grafana/api/v1/rules/{Namespace}:[GET POST DELETE] /ruler/grafana/api/v1/rules/{Namespace}/export:[POST] /ruler/grafana/api/v1/rules/{Namespace}/{Groupname}:[GET DELETE]```, and this change only adds a new method, not a new path. I wouldn't expect this number to increase for this change.

**Comment by joeblubaugh** [comment]
Tests are looking green. @yuri-tceretian could you merge when you're available?

**Pull request merged**